ddtrace/appsec/_api_security/api_manager.py

```diff
@@ -146,29 +147,37 @@ def _schema_callback(self, env):
         except Exception:
             self._log_limiter.limit(log.debug, "Failed to enrich request span with headers", exc_info=True)
 
-        for address, meta_name, transform in self.COLLECTED:
-            value = env.waf_addresses.get(address, _sentinel)
+        waf_payload = {}
+        for address, _, transform in self.COLLECTED:
+            value = env.waf_addresses.get(SPAN_DATA_NAMES[address], _sentinel)
             if value is _sentinel:
                 log.debug("no value for %s", address)
                 continue
-            json_serialized = ()
-            b64_gzip_content = b""
-            try:
-                if transform is not None:
-                    value = transform(value)
-                json_serialized = get_json_schema(value)
-                b64_gzip_content = base64.b64encode(gzip.compress(json_serialized.encode())).decode()
-                if len(b64_gzip_content) >= MAX_SPAN_META_VALUE_LEN:
-                    raise TooLargeSchemaException
-                root._meta[meta_name] = b64_gzip_content
-            except Exception as e:
-                self._schema_meter.increment("errors", tags={"exc": e.__class__.__name__, "address": address})
-                self._log_limiter.limit(
-                    log.warning,
-                    "Failed to get schema from %r [schema length=%d]:\n%s",
-                    address,
-                    len(b64_gzip_content),
-                    repr(value)[:256],
-                    exc_info=True,
-                )
+            if transform is not None:
+                value = transform(value)
+            waf_payload[address] = value
+        if waf_payload:
+            waf_payload["SETTINGS"] = {"extract-schema": True}
```

Comment:
Can we add a basic sampling? Like https://github.com/DataDog/dd-trace-py/blob/c5dfcec45ae21483171c0ac148f94e61ade0b4d2/ddtrace/sampler.py#L83.